src/hott/types/2_adj/adj.lean

```diff
@@ -0,0 +1,165 @@
+      : is_contr(Σ(l : linv u.1 f), Π(x : A), rcoh f ⟨u.1, (l, u.2)⟩ x) :=
+    begin
+      apply @is_trunc.is_trunc_equiv_closed_rev _ (Σ(l : linv u.1 f), Π(x : A), u.2 (f x) = ap f (l x)) -2,
+      apply sigma.sigma_equiv_sigma_right, intro η,
+      apply pi.pi_equiv_pi_right, intro x,
+      apply (eq_equiv_eq_symm (u.2 (f x)) (ap f (η x)))⁻¹ᵉ,
+      exact is_equiv.is_contr_right_coherence f u
+    end
+
+    @[hott, instance] def is_contr_lcoh (f : A → B) [H : is_equiv f] (u : Σ(g : B → A), linv g f)
+      : is_contr(Σ(r : rinv u.1 f), Π(y : B), lcoh f ⟨u.1, (u.2, r)⟩ y) :=
+    begin
+      apply is_trunc.is_trunc_equiv_closed_rev -2
+        (sigma.sigma_pi_equiv_pi_sigma (λy, λr : f (u.1 y) = y, u.2 (u.1 y) = ap u.1 r)),
+      apply is_trunc.is_trunc_equiv_closed -2 (pi.pi_equiv_pi_right (
+        λy : B, @fiber.fiber_eq_equiv _ _ u.1 (u.1 y) 
+        (fiber.mk (f (u.1 y)) (u.2 (u.1 y))) (fiber.mk y rfl) 
+      )),
+      have : f⁻¹ᶠ = u.1 := 
+        (@is_trunc.eq_of_is_contr _ (is_contr_linv f) ⟨f⁻¹ᶠ, is_equiv.left_inv f⟩ u)..1,
+      have : is_equiv u.1 := is_equiv.is_equiv_eq_closed this,
+      exact @pi.is_trunc_pi _ _ -2 (λy, 
+        @is_trunc.is_contr_eq _ (@is_equiv.is_contr_fiber_of_is_equiv _ _ u.1 this (u.1 y)) 
+        (fiber.mk (f (u.1 y)) (u.2 (u.1 y)))
+        (fiber.mk y rfl))
+    end
+
+    -- Left half-adjoint equivalence is a mere proposition
+    @[hott, instance] def is_prop_hadj_l (f : A → B) : is_prop (is_hadj_l f) :=
+    begin
+      apply is_trunc.is_prop_of_imp_is_contr, intro h,
+      apply is_trunc.is_trunc_equiv_closed_rev -2 (
+        sigma.sigma_assoc_equiv (λu: Σ(g : B → A), linv g f, 
+        Σ(r : rinv u.1 f), Π(y : B), lcoh f ⟨u.1, (u.2, r)⟩ y)
+      ),
+      have H : is_equiv f := is_equiv.adjointify f h.1 h.2.2.1 h.2.1,
+      apply is_trunc.is_trunc_equiv_closed_rev -2 (
+        @sigma.sigma_equiv_of_is_contr_left _ (λu : Σ(g : B → A), linv g f, 
+          Σ(r : rinv u.1 f), Π(y : B), lcoh f ⟨u.1, (u.2, r)⟩ y)
+            (@is_contr_linv _ _ f H)),
+      exact @is_contr_lcoh _ _ f H _
+    end
+
+    -- Left half-adjoint equivalence is equivalent to is_equiv
+    @[hott] def is_hadj_l_equiv_is_equiv (f : A → B) : is_hadj_l f ≃ is_equiv f :=
+    is_trunc.equiv_of_is_prop 
+      (λh : is_hadj_l f, is_equiv.adjointify f h.1 h.2.2.1 h.2.1)
+      (λH : is_equiv f, 
+      have Σ(r : rinv H.inv f), Π(y : B), lcoh f ⟨H.inv, (H.left_inv, r)⟩ y, 
+        from @is_trunc.center _ (@is_contr_lcoh _ _ f H ⟨H.inv, H.left_inv⟩),
+      ⟨H.inv, ⟨H.left_inv, this⟩⟩)
+
+    -- Promoting qinv to a left half-adjoint
+    @[hott] def adjointify_left (f : A → B) (inv : B → A) (left_inv : linv inv f) (right_inv : rinv inv f)
+      : is_hadj_l f :=
+    is_equiv.inv (is_hadj_l_equiv_is_equiv f) (is_equiv.adjointify f inv right_inv left_inv)
+
+    -- Definition of a full-adjoint equivalence
+    @[hott] def adj (f : A → B) :=
```

Comment:
It might be useful to give the projections of this definition a name, like 
```
def adj.ε (h : adj f) := h.2.2.1
```
(and same for the other projections) 

If you do that, whenever you have `(h : adj f)` you can write `h.ε` for the corresponding projection.
This should make the file `two_adj.lean` more readable.